### ISSUE TITLE
`ZipEntryTest.setEntryDosTime` can expose segmentation fault

### ISSUE THREAD

**Issue by WojciechMazur** [scala-native/scala-native#3816]
During latest local development I've observed some issues with `ZipEntryTest.setEntryDosTime` with either assertion failing (due to incorrect value) or in the segmentation fault as seen in some nightly builds: 
- https://github.com/scala-native/scala-native/actions/runs/8181799008/job/22372119511#step:4:411 
- https://github.com/scala-native/scala-native/actions/runs/8181799008/job/22372119100#step:4:471
- https://github.com/scala-native/scala-native/actions/runs/8181799008/job/22372119511#step:4:448
- https://github.com/scala-native/scala-native/actions/runs/8181813007/job/22373339673#step:5:792

**Comment by LeeTibbert** [comment]
You caught the devil in the act, and in linux to boot.

TL; DR;  I'll disable the Test, I believe the underlying code is correct.  I can also disable the underlying
              code if that might give peace of mind for the imminent 0.5.0 release.  Don't want random
              SegFaults rattling around, especially with my name on them.

Sorry this is happening.  I believe this is the SN ecosystem timing glitch I mentioned in the
PR:
```
I believe that this PR is bringing out
a long standing timing issue. The "expected" value reported in assert statements is orders of magnitude
too large. The value reported by getTime() is what JVM reports for the "1999 May 25 17:20:06" date
in the file written by ZipEntryTest. Now to catch the timing glitch in the act.
```

The exact same call is being made in two places, once in a `@Before` (before Class) and a second time
in the actual Test.  The first call sets up an "Expected" value.  It is my understanding that 
`@Before` has (or is supposed to have) a "happens before" relation to any "Test" in the companion class.

In the logs, the actual (rightmost) value returned for the second call is correct. Sometimes, the first (leftmost)
call is off by t0 or so orders of magnitude (appears to be total garbage).

I struggled with this over the weekend and had thought I had fixed it (by synchronizing call to `mktime()`).  
I will study this some more.

I'll disable `EntryTest`  for now and work privately.  I would like to uncover the underlying defect, either in
my understanding of the correct timing or in SN's execution.   The fields in play are `Short`s, so that may
be a factor.   I think the presenting problem is in the Test suite and that the underlying code is OK.

**Comment by LeeTibbert** [comment]
I was seeing the reported problem on macOS with JVM acceptance runs. I  applied some "Big Hammer" synchronization
which made the problem disappear.  The problem never showed in "normal" CI test-runtime.

I have to see how the "JVM Acceptance" environment and the two "special" ones reported here differ from
the "normal" CI test-runtime.  I better change my Thanksgiving plans.
